entity-framework/core/get-started/winforms.md

```diff
@@ -0,0 +1,318 @@
+
+> [!TIP]
+> When using Windows Forms, the `ObservableCollectionListSource`, which implements `IListSource`, can be used for collection navigations. This is not necessary, but does improve the two-way data binding experience.
+
+## Define the DbContext
+
+In EF Core, a class derived from `DbContext` is used to configure entity types in a model and act as a session for interacting with the database. In the simplest case, a `DbContext` class:
+
+- Contains `DbSet` properties for each entity type in the model.
+- Overrides the `OnConfiguring` method to configure the database provider and connection string to use. See [Configuring a DbContext](xref:core/dbcontext-configuration/index) for more information.
+
+In this case, the DbContext class also overrides the `OnModelCreating` method to provide some sample data for the application.
+
+Add a new `ProductsContext.cs` class to the project with the following code:
+
+[!code-csharp[](../../../samples/core/WinForms/GetStartedWinForms/GetStartedWinForms/ProductsContext.cs)]
+
+Make sure to **build the solution** at this point.
+
+## Adding controls to the form
+
+The application will show a list of categories and a list of products. When a category is selected in the first list, then the second list will change to show products for that category. These lists can be modified to add, remove, or edit products and categories, and these changes can be saved to the SQLite database by clicking a **Save** button.
+
+1. Change the name of the main form from `Form1` to `MainForm`.
+
+   ![Rename Form1 to MainForm](_static/winforms-mainform.png)
+
+2. And change the title to "Products and Categories".
+
+   ![Title MainForm as "Products and Categories"](_static/winforms-products-and-categories.png)
+
+3. Using the **Toolbox**, add two `DataGridView` controls, arranged next to each other.
+
+   ![Add DataGridView](_static/winforms-datagrid-view.png)
+
+4. In the **Properties** for the first `DataGridView`, change the **Name** to `dataGridViewCategories`.
+5. In the **Properties** for the second `DataGridView`, change the **Name** to `dataGridViewProducts`.
+6. Also using the **Toolbox**, add a `Button` control.
+7. Name the button `buttonSave` and give it the text "Save". The form should look something this this:
+
+   ![Form layout](_static/winforms-form-layout.png)
+
+## Data binding
+
+The next step is to connect the `Product` and `Category` types from the model to the `DataGridView` controls. This will bind the data loaded by EF Core to the controls, such that the entities tracked by EF Core are kept in sync with those displayed in the controls.
+
+1. Click the **Design Binding Picker** on the first `DataGridView`. This is the tiny button at the top-right corner of the control.
```

Comment:
A bit terminology: The "tiny button" is the _Designer Action Glyph_. You open the _Action List_ with it (which is what you described as the Design Binding Picker), and then you get access to the Design Binding Picker, by accessing the type editor of the **DataSource**. A Type Editor is either the value picker provided by the drop-down of a property in the Property Browser or a modal dialog to edit the type - also initiated through a property in that browser (then with the three-dotted button). The most frequently used Typed Editors of a control's properties can - like here - also called via its Action List. The _Design Binding Picker_ is technically only the content of that drop down in this case. (Terminology in WinForms is confusing, I know - we always use the euphemism "WinForm's terminology has historically grown"! 😄)